### PR TITLE
fix: option component accept number on name prop

### PR DESCRIPTION
**Pull request by yvmunayev**
fix: #1692

## Changes proposed in this PR:
- fix: option component accept number on name prop

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/Option/index.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class OptionItem extends Component {
     componentDidMount() {
         const { disabled, variant, name } = this.props;
         const isHeader = variant === 'header';
-        if (disabled || isHeader || typeof name !== 'string') {
+        if (disabled || isHeader || (typeof name !== 'string' && typeof name !== 'number')) {
```
Comment (LeandroTorresSicilia) [inline src/components/Option/index.js:31]: we should add some test here

File: src/components/Option/index.js [hunk 2]
```diff
@@ -201,7 +201,7 @@ Option.propTypes = {
     /** Text of the PicklistOption. */
     label: PropTypes.string,
     /** The name of the PicklistOption. */
-    name: PropTypes.string,
+    name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (LeandroTorresSicilia) [inline src/components/Option/index.js:204]: we need to update the index.d.ts files, also for multiselect component

File: src/components/Picklist/index.js [hunk 3]
```diff
@@ -299,7 +299,7 @@ Picklist.propTypes = {
     value: PropTypes.oneOfType([
         PropTypes.shape({
             label: PropTypes.string,
-            name: PropTypes.string,
+            name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (LeandroTorresSicilia) [inline src/components/Picklist/index.js:302]: same ^^ update index.d.ts

File: src/components/Option/readme.md [hunk 4]
```diff
@@ -69,3 +69,39 @@ initialState = { value: null };
...
+
+initialState = { value: { name: 1, label: 'Experimental Building' } };
+
+<div className="rainbow-m-bottom_xx-large rainbow-p-bottom_xx-large">
+    <GlobalHeader
+        src="images/user/user2.jpg"
+        className="rainbow-p-bottom_xx-large rainbow-m-bottom_xx-large"
+    >
+        <div className="rainbow-flex rainbow-align_right">
+            <Picklist
```
Comment (TahimiLeonBravo) [inline src/components/Option/readme.md:91]: @yvmunayev I think we should modify one of the existing examples instead add one new

**Pull request merged**